test/aws-redshiftserverless/namespace.test.ts

```diff
@@ -0,0 +1,243 @@
+    });
+
+    test.each([0, 3654])(
+      'throws when finalSnapshotRetentionPeriod is invalid, got %d',
+      finalSnapshotRetentionPeriod => {
+        expect(() => {
+          new Namespace(stack, 'Namespace', {
+            finalSnapshotName: 'my-final-snapshot',
+            finalSnapshotRetentionPeriod,
+          });
+        }).toThrow(
+          `\`finalSnapshotRetentionPeriod\` must be between 1 and 3653, got: ${finalSnapshotRetentionPeriod}.`,
+        );
+      },
+    );
+  });
+
+  describe('validateDefaultIamRole test', () => {
+    test('throws when defaultIamRole role is not included in iamRoles', () => {
+      expect(() => {
+        const defaultIamRole = new Role(stack, 'DefaultRole', {
+          assumedBy: new ServicePrincipal('redshift.amazonaws.com'),
+        });
+
+        const anotherRole = new Role(stack, 'AnotherRole', {
+          assumedBy: new ServicePrincipal('redshift.amazonaws.com'),
+        });
+
+        new Namespace(stack, 'Namespace', {
+          defaultIamRole,
+          iamRoles: [anotherRole],
+        });
+      }).toThrow('`defaultIamRole` must be included in `iamRoles`.');
+    });
+  });
+
+  describe('validateNamespaceName test', () => {
+    test.each(['UpperName', 'invalid$name'])('throws when namespaceName is invalid , got %s', namespaceName => {
+      expect(() => {
+        new Namespace(stack, 'Namespace', {
+          namespaceName,
+        });
+      }).toThrow(
+        `\`namespaceName\` must consist only of lowercase alphanumeric characters or hyphens, got: ${namespaceName}.`,
+      );
+    });
+
+    test.each(['a'.repeat(2), 'a'.repeat(65)])('throws when namespaceName length is invalid, got %s', namespaceName => {
+      expect(() => {
+        new Namespace(stack, 'Namespace', {
+          namespaceName,
+        });
+      }).toThrow(`\`namespaceName\` must be between 3 and 64 characters, got: ${namespaceName.length} characters.`);
+    });
+  });
+
+  describe('test addIamRole method', () => {
+    test('add IAM Role after creation', () => {
+      const namespace = new Namespace(stack, 'Namespace', {});
+
+      const addRole = new Role(stack, 'AddRole', {
+        assumedBy: new ServicePrincipal('redshift.amazonaws.com'),
+      });
+
+      namespace.addIamRole(addRole);
+
+      Template.fromStack(stack).hasResourceProperties('AWS::RedshiftServerless::Namespace', {
+        IamRoles: [stack.resolve(addRole.roleArn)],
+      });
+    });
+
+    test('throws when an adding IAM role is already attached to the namespace', () => {
+      const addRole = new Role(stack, 'AddRole', {
+        assumedBy: new ServicePrincipal('redshift.amazonaws.com'),
+      });
+
+      expect(() => {
+        const namespace = new Namespace(stack, 'Namespace', {
+          iamRoles: [addRole],
+        });
+        namespace.addIamRole(addRole);
+      }).toThrow(
+        `An adding IAM Role is already attached to the namespace, name: ${addRole.roleName}, ARN: ${addRole.roleArn}.`,
```

Comment:
see: https://github.com/open-constructs/aws-cdk-library/pull/43#discussion_r1780074005

Reply:
I've modified it to only include ARN.